Index: src/Guile/Glue/build-glue-basic.scm
===================================================================
--- src/Guile/Glue/build-glue-basic.scm	(revision 4584)
+++ src/Guile/Glue/build-glue-basic.scm	(working copy)
@@ -40,6 +40,7 @@
   (eval-system eval_system (string string))
   (var-eval-system var_eval_system (string string))
   (get-locale-language get_locale_language (string))
+  (get-locale-encoding get_locale_encoding (string))  
   (texmacs-time texmacs_time (int))
   (texmacs-memory mem_used (int))
   (bench-print bench_print (void string))
Index: src/Guile/Glue/glue_basic.cpp
===================================================================
--- src/Guile/Glue/glue_basic.cpp	(revision 4584)
+++ src/Guile/Glue/glue_basic.cpp	(working copy)
@@ -257,6 +257,15 @@
 }
 
 SCM
+tmg_get_locale_encoding () {
+  // SCM_DEFER_INTS;
+  string out= get_locale_encoding ();
+  // SCM_ALLOW_INTS;
+
+  return string_to_scm (out);
+}
+
+SCM
 tmg_texmacs_time () {
   // SCM_DEFER_INTS;
   int out= texmacs_time ();
@@ -4992,6 +5001,7 @@
   scm_new_procedure ("eval-system", (FN) tmg_eval_system, 1, 0, 0);
   scm_new_procedure ("var-eval-system", (FN) tmg_var_eval_system, 1, 0, 0);
   scm_new_procedure ("get-locale-language", (FN) tmg_get_locale_language, 0, 0, 0);
+  scm_new_procedure ("get-locale-encoding", (FN) tmg_get_locale_encoding, 0, 0, 0);
   scm_new_procedure ("texmacs-time", (FN) tmg_texmacs_time, 0, 0, 0);
   scm_new_procedure ("texmacs-memory", (FN) tmg_texmacs_memory, 0, 0, 0);
   scm_new_procedure ("bench-print", (FN) tmg_bench_print, 1, 0, 0);
Index: src/System/Language/language.hpp
===================================================================
--- src/System/Language/language.hpp	(revision 4584)
+++ src/System/Language/language.hpp	(working copy)
@@ -128,6 +128,7 @@
 string locale_to_language (string s);
 string language_to_locale (string s);
 string get_locale_language ();
+string get_locale_encoding ();
 #if defined(__MINGW__) || defined(__MINGW32__) || defined(OS_WIN32)
 namespace win32 {
   string get_date (string lan, string fm);
Index: src/System/Language/text_language.cpp
===================================================================
--- src/System/Language/text_language.cpp	(revision 4584)
+++ src/System/Language/text_language.cpp	(working copy)
@@ -278,6 +278,20 @@
 #endif
 }
 
+string
+get_locale_encoding () {
+  string lan = get_locale_language( );
+#if defined(_WIN32) || defined(__WIN32__)
+  if (lan == "chinese") return "GBK";
+  if (lan == "taiwanese") return "BIG5";
+  if (lan == "japanese") return "CP932";
+  if (lan == "korean") return "CP949";
+  return "UTF-8";
+#else
+  return "UTF-8";
+#endif
+}
+
 /******************************************************************************
 * Getting a formatted date
 ******************************************************************************/
Index: TeXmacs/progs/texmacs/menus/file-menu.scm
===================================================================
--- TeXmacs/progs/texmacs/menus/file-menu.scm	(revision 4584)
+++ TeXmacs/progs/texmacs/menus/file-menu.scm	(working copy)
@@ -34,7 +34,7 @@
 
 (tm-menu (file-list-menu l)
   (for (name l)
-    (let* ((short-name (url->string (url-tail name))))
+    (let* ((short-name (string-convert (url->string (url-tail name)) (get-locale-encoding) "Cork")))
       ((balloon (eval short-name) (eval name)) (load-buffer name)))))
 
 (tm-define (recent-file-menu)
